Fix to avoid nil reference bug

ambiknai opened the pull request:
In unmanaged clusters, we were hitting with Nil reference error since "clusterID" is nil.
This PR is to fix the bug.
Once PR is merged, master branch will be picked to test on both managed and unmanaged clusters.
A release will be done only after verifying the fix

arahamad reviewed (approved):
lgtm, 
please verify on managed and un-managed cluster

sameshai reviewed (approved):
Paste result of managed/unmanaged cluster rest LGTM

ambiknai commented:
`Test`

```
git diff go.mod
diff --git a/go.mod b/go.mod
index 3c1041f..651a099 100644
--- a/go.mod
+++ b/go.mod
@@ -3,14 +3,14 @@ module github.com/kubernetes-sigs/ibm-vpc-block-csi-driver
 go 1.16
 
 require (
-       github.com/IBM/ibm-csi-common v1.0.0-beta3
+       github.com/IBM/ibm-csi-common v1.0.0-beta5.0.20210920083614-58a8a94ea1b8
        github.com/IBM/ibmcloud-volume-interface v1.0.0-beta7
+       github.com/IBM/ibmcloud-volume-vpc v1.0.0-beta9 // indirect
        github.com/container-storage-interface/spec v1.3.0
        github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
        github.com/google/uuid v1.1.2
        github.com/kubernetes-csi/csi-test/v4 v4.2.0
        github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
-       github.com/pierrre/gotestcover v0.0.0-20160517101806-924dca7d15f0 // indirect
        github.com/prometheus/client_golang v1.7.1
        github.com/stretchr/testify v1.6.1
        go.uber.org/zap v1.15.0

```

ambiknai commented:
`unmanaged cluster`

```
kubectl get nodes -o wide
NAME                   STATUS   ROLES           AGE   VERSION           INTERNAL-IP    EXTERNAL-IP   OS-IMAGE   KERNEL-VERSION                CONTAINER-RUNTIME
sat-ocp-workernode01   Ready    master,worker   18d   v1.20.0+558d959   10.242.0.5     <none>        Red Hat    3.10.0-1160.36.2.el7.x86_64   cri-o://1.20.4-8.rhaos4.7.git74c6592.el7
sat-ocp-workernode02   Ready    master,worker   18d   v1.20.0+558d959   10.242.64.6    <none>        Red Hat    3.10.0-1160.36.2.el7.x86_64   cri-o://1.20.4-8.rhaos4.7.git74c6592.el7
sat-ocp-workernode03   Ready    master,worker   18d   v1.20.0+558d959   10.242.128.5   <none>        Red Hat    3.10.0-1160.36.2.el7.x86_64   cri-o://1.20.4-8.rhaos4.7.git74c6592.el7
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# 
```

```
bash deploy/kubernetes/driver/kubernetes/deploy-vpc-block-driver.sh stage gen2
This will install 'stage' version of vpc csi driver for 'gen2' !
serviceaccount/ibm-vpc-block-controller-sa configured
serviceaccount/ibm-vpc-block-node-sa configured
clusterrole.rbac.authorization.k8s.io/vpc-block-driver-registrar-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-external-attacher-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-external-resizer-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-provisioner-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-driver-registrar-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-external-attacher-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-external-resizer-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-provisioner-binding unchanged
configmap/ibm-vpc-block-csi-configmap unchanged
secret/storage-secret-store unchanged
statefulset.apps/ibm-vpc-block-csi-controller created
daemonset.apps/ibm-vpc-block-csi-node created
Warning: storage.k8s.io/v1beta1 CSIDriver is deprecated in v1.19+, unavailable in v1.22+; use storage.k8s.io/v1 CSIDriver
csidriver.storage.k8s.io/vpc.block.csi.ibm.io unchanged

```

```
# oc describe pods -n kube-system ibm-vpc-block-csi-controller-0 ibm-vpc-block-csi-node-52rp8
Name:               ibm-vpc-block-csi-controller-0
Namespace:          kube-system
Priority:           0
PriorityClassName:  <none>
Node:               sat-ocp-workernode03/10.242.128.5
Start Time:         Mon, 20 Sep 2021 02:04:52 -0700
Labels:             app=ibm-vpc-block-csi-driver
                    controller-revision-hash=ibm-vpc-block-csi-controller-59b789c4d6
                    statefulset.kubernetes.io/pod-name=ibm-vpc-block-csi-controller-0
Annotations:        cni.projectcalico.org/podIP: 172.30.105.68/32
                    cni.projectcalico.org/podIPs: 172.30.105.68/32
                    k8s.v1.cni.cncf.io/network-status:
                      [{
                          "name": "",
                          "ips": [
                              "172.30.105.68"
                          ],
                          "default": true,
                          "dns": {}
                      }]
                    k8s.v1.cni.cncf.io/networks-status:
                      [{
                          "name": "",
                          "ips": [
                              "172.30.105.68"
                          ],
                          "default": true,
                          "dns": {}
                      }]
                    prometheus.io/path: /metrics
                    prometheus.io/port: 9080
                    prometheus.io/scrape: true
Status:             Running
IP:                 172.30.105.68
Controlled By:      StatefulSet/ibm-vpc-block-csi-controller
Containers:
  csi-provisioner:
    Container ID:  cri-o://7be5772df36a49ea00ea69b7112d2b957826f7627ff7f95857ee00398f61cb36
    Image:         quay.io/k8scsi/csi-provisioner:v1.6.0
    Image ID:      quay.io/k8scsi/csi-provisioner@sha256:78e3393f5fd5ff6c1e5dada2478cfa456fb7164929e573cf9a87bf6532730679
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=$(ADDRESS)
      --timeout=600s
      --feature-gates=Topology=true
    State:          Running
      Started:      Mon, 20 Sep 2021 02:04:55 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-klmcb (ro)
  csi-attacher:
    Container ID:  cri-o://71e215f765fb2deb4bb5be6c79545753eefd4fccc149cb41ac42d2b9ac3b2e14
    Image:         quay.io/k8scsi/csi-attacher:v2.2.0
    Image ID:      quay.io/k8scsi/csi-attacher@sha256:2ffa647e8107cfd39e5f464e738dce014c9f5e51b108da36c3ab621048d0bbab
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=/csi/csi.sock
      --timeout=900s
    State:          Running
      Started:      Mon, 20 Sep 2021 02:04:55 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:        10m
      memory:     20Mi
    Environment:  <none>
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-klmcb (ro)
  liveness-probe:
    Container ID:  cri-o://5100603f96a0fda513302b338ee59201509a4eecd33149e6abe28fc72cf13973
    Image:         quay.io/k8scsi/livenessprobe:v2.0.0
    Image ID:      quay.io/k8scsi/livenessprobe@sha256:ff8c925b509ebfcf0487e877a2722f87c01632f8d0c5c7eda94c66959542d290
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=/csi/csi.sock
    State:          Running
      Started:      Mon, 20 Sep 2021 02:04:56 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     50m
      memory:  50Mi
    Requests:
      cpu:        5m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-klmcb (ro)
  iks-vpc-block-driver:
    Container ID:  cri-o://2cc5fc30243a46b4da5fa5c376b14098e3f8acd6986ae574a71348a3eaa43e49
    Image:         ambikanair/ibm-vpc-block-csi-driver:nilfix
    Image ID:      docker.io/ambikanair/ibm-vpc-block-csi-driver@sha256:618799307d2b0dd3a0ea52ef8a83dacc15e681d2d6becc4798006bf58e01471e
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --v=5
      --endpoint=$(CSI_ENDPOINT)
      --lock_enabled=false
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:02 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     500m
      memory:  500Mi
    Requests:
      cpu:     50m
      memory:  100Mi
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=10s #success=1 #failure=5
    Environment Variables from:
      ibm-vpc-block-csi-configmap  ConfigMap  Optional: false
    Environment:
      POD_NAME:       ibm-vpc-block-csi-controller-0 (v1:metadata.name)
      POD_NAMESPACE:  kube-system (v1:metadata.namespace)
    Mounts:
      /csi from socket-dir (rw)
      /etc/storage_ibmc from customer-auth (ro)
      /etc/storage_ibmc/cluster_info from cluster-info (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-klmcb (ro)
  csi-resizer:
    Container ID:  cri-o://1b18e339813c2a29d7874572fe831e15b1015bbd0cf7676b0fa7dd5e8f3f7c4e
    Image:         quay.io/k8scsi/csi-resizer:v1.0.0
    Image ID:      quay.io/k8scsi/csi-resizer@sha256:83655a67f84ac500b67fd0e7084d1eaa9828321d179269171aa0c9bc35ca8ef9
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=$(ADDRESS)
      --timeout=600s
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:06 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-klmcb (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  socket-dir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:  
  customer-auth:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  storage-secret-store
    Optional:    false
  cluster-info:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cluster-info
    Optional:  false
  ibm-vpc-block-controller-sa-token-klmcb:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ibm-vpc-block-controller-sa-token-klmcb
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason          Age   From                           Message
  ----    ------          ----  ----                           -------
  Normal  Scheduled       79s   default-scheduler              Successfully assigned kube-system/ibm-vpc-block-csi-controller-0 to sat-ocp-workernode03
  Normal  AddedInterface  77s   multus                         Add eth0 [172.30.105.68/32]
  Normal  Pulling         77s   kubelet, sat-ocp-workernode03  Pulling image "quay.io/k8scsi/csi-provisioner:v1.6.0"
  Normal  Pulled          76s   kubelet, sat-ocp-workernode03  Container image "quay.io/k8scsi/csi-attacher:v2.2.0" already present on machine
  Normal  Pulled          76s   kubelet, sat-ocp-workernode03  Successfully pulled image "quay.io/k8scsi/csi-provisioner:v1.6.0" in 1.083157665s
  Normal  Started         76s   kubelet, sat-ocp-workernode03  Started container csi-provisioner
  Normal  Created         76s   kubelet, sat-ocp-workernode03  Created container csi-attacher
  Normal  Started         76s   kubelet, sat-ocp-workernode03  Started container csi-attacher
  Normal  Pulled          76s   kubelet, sat-ocp-workernode03  Container image "quay.io/k8scsi/livenessprobe:v2.0.0" already present on machine
  Normal  Created         76s   kubelet, sat-ocp-workernode03  Created container csi-provisioner
  Normal  Pulling         75s   kubelet, sat-ocp-workernode03  Pulling image "ambikanair/ibm-vpc-block-csi-driver:nilfix"
  Normal  Started         75s   kubelet, sat-ocp-workernode03  Started container liveness-probe
  Normal  Created         75s   kubelet, sat-ocp-workernode03  Created container liveness-probe
  Normal  Pulled          69s   kubelet, sat-ocp-workernode03  Successfully pulled image "ambikanair/ibm-vpc-block-csi-driver:nilfix" in 5.637740055s
  Normal  Created         69s   kubelet, sat-ocp-workernode03  Created container iks-vpc-block-driver
  Normal  Started         69s   kubelet, sat-ocp-workernode03  Started container iks-vpc-block-driver
  Normal  Pulling         69s   kubelet, sat-ocp-workernode03  Pulling image "quay.io/k8scsi/csi-resizer:v1.0.0"
  Normal  Pulled          65s   kubelet, sat-ocp-workernode03  Successfully pulled image "quay.io/k8scsi/csi-resizer:v1.0.0" in 3.498627555s
  Normal  Created         65s   kubelet, sat-ocp-workernode03  Created container csi-resizer
  Normal  Started         65s   kubelet, sat-ocp-workernode03  Started container csi-resizer


Name:               ibm-vpc-block-csi-node-52rp8
Namespace:          kube-system
Priority:           0
PriorityClassName:  <none>
Node:               sat-ocp-workernode02/10.242.64.6
Start Time:         Mon, 20 Sep 2021 02:04:53 -0700
Labels:             app=ibm-vpc-block-csi-driver
                    controller-revision-hash=6469b6bcf6
                    pod-template-generation=1
Annotations:        cni.projectcalico.org/podIP: 172.30.112.53/32
                    cni.projectcalico.org/podIPs: 172.30.112.53/32
                    k8s.v1.cni.cncf.io/network-status:
                      [{
                          "name": "",
                          "ips": [
                              "172.30.112.53"
                          ],
                          "default": true,
                          "dns": {}
                      }]
                    k8s.v1.cni.cncf.io/networks-status:
                      [{
                          "name": "",
                          "ips": [
                              "172.30.112.53"
                          ],
                          "default": true,
                          "dns": {}
                      }]
                    prometheus.io/path: /metrics
                    prometheus.io/port: 9080
                    prometheus.io/scrape: true
Status:             Running
IP:                 172.30.112.53
Controlled By:      DaemonSet/ibm-vpc-block-csi-node
Containers:
  iks-vpc-block-node-driver:
    Container ID:  cri-o://9be0eb3c462898aa4e45ca99ab75158196cb84453635817b2a8da474d47acd68
    Image:         ambikanair/ibm-vpc-block-csi-driver:nilfix
    Image ID:      docker.io/ambikanair/ibm-vpc-block-csi-driver@sha256:618799307d2b0dd3a0ea52ef8a83dacc15e681d2d6becc4798006bf58e01471e
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --v=5
      --endpoint=unix:/csi/csi.sock
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:00 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     200m
      memory:  250Mi
    Requests:
      cpu:     20m
      memory:  50Mi
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=10s #success=1 #failure=5
    Environment Variables from:
      ibm-vpc-block-csi-configmap  ConfigMap  Optional: false
    Environment:
      KUBE_NODE_NAME:   (v1:spec.nodeName)
    Mounts:
      /csi from plugin-dir (rw)
      /dev from device-dir (rw)
      /etc/storage_ibmc from customer-auth (ro)
      /etc/storage_ibmc/cluster_info from cluster-info (ro)
      /etc/udev from etcudevpath (rw)
      /lib/udev from libudevpath (rw)
      /run/udev from runudevpath (rw)
      /sys from syspath (rw)
      /var/data/kubelet from kubelet-data-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-node-sa-token-chfxc (ro)
  csi-driver-registrar:
    Container ID:  cri-o://c00a9db856dd0727f79a5fd3d7006b816153697a5121cc4e34c33c0642551fb5
    Image:         quay.io/k8scsi/csi-node-driver-registrar:v1.2.0
    Image ID:      quay.io/k8scsi/csi-node-driver-registrar@sha256:89cdb2a20bdec89b75e2fbd82a67567ea90b719524990e772f2704b19757188d
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=$(ADDRESS)
      --kubelet-registration-path=$(DRIVER_REGISTRATION_SOCK)
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:03 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:                   /csi/csi.sock
      DRIVER_REGISTRATION_SOCK:  /var/lib/kubelet/csi-plugins/vpc.block.csi.ibm.io/csi.sock
      KUBE_NODE_NAME:             (v1:spec.nodeName)
    Mounts:
      /csi from plugin-dir (rw)
      /registration from registration-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-node-sa-token-chfxc (ro)
  liveness-probe:
    Container ID:  cri-o://03afb7a8c8cde1c7dda0ca7cec5a3fb24c6464b30494a750b85ada70b0136388
    Image:         quay.io/k8scsi/livenessprobe:v2.0.0
    Image ID:      quay.io/k8scsi/livenessprobe@sha256:ff8c925b509ebfcf0487e877a2722f87c01632f8d0c5c7eda94c66959542d290
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=/csi/csi.sock
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:03 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     50m
      memory:  50Mi
    Requests:
      cpu:        5m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /csi from plugin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-node-sa-token-chfxc (ro)
  csi-resizer:
    Container ID:   cri-o://e75ab59b14c5bf9854f1937be223ba5f1ff9079a81c70bc30cb45bea0d7b03bd
    Image:          quay.io/k8scsi/csi-resizer:v1.0.0
    Image ID:       quay.io/k8scsi/csi-resizer@sha256:83655a67f84ac500b67fd0e7084d1eaa9828321d179269171aa0c9bc35ca8ef9
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Mon, 20 Sep 2021 02:05:04 -0700
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-node-sa-token-chfxc (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  registration-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins_registry/
    HostPathType:  Directory
  kubelet-data-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/data/kubelet
    HostPathType:  Directory
  plugin-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/csi-plugins/vpc.block.csi.ibm.io/
    HostPathType:  DirectoryOrCreate
  device-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /dev
    HostPathType:  Directory
  etcudevpath:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/udev
    HostPathType:  Directory
  runudevpath:
    Type:          HostPath (bare host directory volume)
    Path:          /run/udev
    HostPathType:  Directory
  libudevpath:
    Type:          HostPath (bare host directory volume)
    Path:          /lib/udev
    HostPathType:  Directory
  syspath:
    Type:          HostPath (bare host directory volume)
    Path:          /sys
    HostPathType:  Directory
  customer-auth:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  storage-secret-store
    Optional:    false
  cluster-info:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cluster-info
    Optional:  false
  ibm-vpc-block-node-sa-token-chfxc:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ibm-vpc-block-node-sa-token-chfxc
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     
Events:
  Type    Reason          Age   From                           Message
  ----    ------          ----  ----                           -------
  Normal  Scheduled       78s   default-scheduler              Successfully assigned kube-system/ibm-vpc-block-csi-node-52rp8 to sat-ocp-workernode02
  Normal  AddedInterface  77s   multus                         Add eth0 [172.30.112.53/32]
  Normal  Pulling         77s   kubelet, sat-ocp-workernode02  Pulling image "ambikanair/ibm-vpc-block-csi-driver:nilfix"
  Normal  Pulled          71s   kubelet, sat-ocp-workernode02  Successfully pulled image "ambikanair/ibm-vpc-block-csi-driver:nilfix" in 5.887941727s
  Normal  Created         71s   kubelet, sat-ocp-workernode02  Created container iks-vpc-block-node-driver
  Normal  Started         71s   kubelet, sat-ocp-workernode02  Started container iks-vpc-block-node-driver
  Normal  Pulling         71s   kubelet, sat-ocp-workernode02  Pulling image "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"
  Normal  Pulled          69s   kubelet, sat-ocp-workernode02  Successfully pulled image "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0" in 2.14026703s
  Normal  Created         68s   kubelet, sat-ocp-workernode02  Created container csi-driver-registrar
  Normal  Started         68s   kubelet, sat-ocp-workernode02  Started container csi-driver-registrar
  Normal  Pulled          68s   kubelet, sat-ocp-workernode02  Container image "quay.io/k8scsi/livenessprobe:v2.0.0" already present on machine
  Normal  Created         68s   kubelet, sat-ocp-workernode02  Created container liveness-probe
  Normal  Started         68s   kubelet, sat-ocp-workernode02  Started container liveness-probe
  Normal  Pulled          68s   kubelet, sat-ocp-workernode02  Container image "quay.io/k8scsi/csi-resizer:v1.0.0" already present on machine
  Normal  Created         67s   kubelet, sat-ocp-workernode02  Created container csi-resizer
  Normal  Started         67s   kubelet, sat-ocp-workernode02  Started container csi-resizer
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# 

```

```
# oc create -f examples/kubernetes/validPVC.yaml
persistentvolumeclaim/csi-block-pvc-one created
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc get pvc
NAME                 STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                AGE
csi-block-pvc-good   Bound    pvc-f681406a-7fa0-4451-b15f-b30e37ff48f8   10Gi       RWO            ibmc-vpc-block-5iops-tier   17d
csi-block-pvc-one    Bound    pvc-0185e150-b82f-4ee6-b2d3-9a80e3a72e42   10Gi       RWO            ibmc-vpc-block-5iops-tier   80s
www-web-0            Bound    pvc-89a01401-3318-437e-9798-79d29e7498cb   25Gi       RWO            ibmc-vpc-block-5iops-tier   12d
www-web-1            Bound    pvc-95571230-9ebb-47a9-835a-dfbadb413662   25Gi       RWO            ibmc-vpc-block-5iops-tier   12d
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc logs -n kube-system ibm-vpc-block-csi-controller-0
Error from server (BadRequest): a container name must be specified for pod ibm-vpc-block-csi-controller-0, choose one of: [csi-provisioner csi-attacher liveness-probe iks-vpc-block-driver csi-resizer]
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc logs -n kube-system ibm-vpc-block-csi-controller-0 -c iks-vpc-block-driver > log.txt
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# vi log.txt
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc get pods -n kube-system | grep block
ibm-vpc-block-csi-controller-0                 5/5     Running   0          7m35s
ibm-vpc-block-csi-node-52rp8                   4/4     Running   0          7m34s
ibm-vpc-block-csi-node-fsxs9                   4/4     Running   0          7m34s
ibm-vpc-block-csi-node-wsh42                   4/4     Running   0          7m34s
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# vi examples/kubernetes/validPOD.yaml 
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc create -f examples/kubernetes/validPOD.yaml
deployment.apps/testpodcsi-block-pvc-one created
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc get pods | grep testpodcsi-block-pvc-one
testpodcsi-block-pvc-one-789769f4c7-lp7md   0/1     ContainerCreating   0          61s
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc get pods | grep testpodcsi-block-pvc-one
testpodcsi-block-pvc-one-789769f4c7-lp7md   1/1     Running   0          113s
root@devmachine1:~/WORKSPACE/src/github.com/kubernetes-sigs/ibm-vpc-block-csi-driver# oc exec -it testpodcsi-block-pvc-one-789769f4c7-lp7md /bin/bash
root@testpodcsi-block-pvc-one-789769f4c7-lp7md:/# ls 
bin  boot  dev	etc  home  lib	lib32  lib64  libx32  media  mnt  myvolumepath	opt  plugins  proc  root  run  sbin  srv  sys  tmp  usr  var
root@testpodcsi-block-pvc-one-789769f4c7-lp7md:/# echo "hi" > myvolumepath/new.txt
root@testpodcsi-block-pvc-one-789769f4c7-lp7md:/# ls myvolumepath
lost+found  new.txt
root@testpodcsi-block-pvc-one-789769f4c7-lp7md:/# cat myvolumepath/new.txt
hi
root@testpodcsi-block-pvc-one-789769f4c7-lp7md:/# 

```

ambiknai commented:
`managed cluster`

```
# export GO111MODULE=on
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# go get github.com/IBM/ibm-csi-common@58a8a94
go: downloading github.com/IBM/ibm-csi-common v1.0.0-beta5.0.20210920083614-58a8a94ea1b8
go get: upgraded github.com/IBM/ibm-csi-common v1.0.0-beta5 => v1.0.0-beta5.0.20210920083614-58a8a94ea1b8
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# git status
On branch testcsi
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   go.mod
	modified:   go.sum

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	pkg/git-commit
	pkg/git-url

no changes added to commit (use "git add" and/or "git commit -a")
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# git diff
diff --git a/go.mod b/go.mod
index 9d82967..d48dd9f 100644
--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver
 go 1.16
 
 require (
-       github.com/IBM/ibm-csi-common v1.0.0-beta5
+       github.com/IBM/ibm-csi-common v1.0.0-beta5.0.20210920083614-58a8a94ea1b8
        github.com/IBM/ibmcloud-volume-interface v1.0.0-beta7
        github.com/container-storage-interface/spec v1.2.0
        github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
diff --git a/go.sum b/go.sum
index ede1f22..cfbae53 100644
--- a/go.sum
+++ b/go.sum
@@ -6,6 +6,8 @@ github.com/IBM-Cloud/ibm-cloud-cli-sdk v0.6.7 h1:eHgfQl6IeSmzWUyiSi13CvoFYsovoyq
 github.com/IBM-Cloud/ibm-cloud-cli-sdk v0.6.7/go.mod h1:RiUvKuHKTBmBApDMUQzBL14pQUGKcx/IioKQPIcRQjs=
 github.com/IBM/ibm-csi-common v1.0.0-beta5 h1:xLh6k8pw8bcDQEe2pUGrFYlpfQJmPrOaarsl0gtUlvA=
 github.com/IBM/ibm-csi-common v1.0.0-beta5/go.mod h1:DswvI7yUVfsLiFYQ/ADjrSqSMijujk3CZpFSbwolQQQ=
+github.com/IBM/ibm-csi-common v1.0.0-beta5.0.20210920083614-58a8a94ea1b8 h1:T9inMxbIIx6dFYCXfZQ24BKfSAnYWXOv3c5wmKo3IR8=
+github.com/IBM/ibm-csi-common v1.0.0-beta5.0.20210920083614-58a8a94ea1b8/go.mod h1:DswvI7yUVfsLiFYQ/ADjrSqSMijujk3CZpFSbwolQQQ=
 github.com/IBM/ibmcloud-volume-interface v1.0.0-beta7 h1:cl1situtbkJwEsRDw1CW2KpJkmznmjzhWWnSPH3TzvM=
 github.com/IBM/ibmcloud-volume-interface v1.0.0-beta7/go.mod h1:2LxqkERml3eBxfRkQniuiRmKkAnl5tfC0ttlP2gupJM=
 github.com/IBM/ibmcloud-volume-vpc v1.0.0-beta11 h1:C3xhQZocc2zhmhyJlB2bVSbowy2sSIgsByBADTk4eIc=
 ```
 
 ```
 # kubectl get nodes -o wide
NAME         STATUS   ROLES    AGE   VERSION        INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION       CONTAINER-RUNTIME
10.249.0.6   Ready    <none>   19d   v1.20.10+IKS   10.249.0.6    10.249.0.6    Ubuntu 18.04.5 LTS   4.15.0-154-generic   containerd://1.4.9
10.249.0.7   Ready    <none>   19d   v1.20.10+IKS   10.249.0.7    10.249.0.7    Ubuntu 18.04.5 LTS   4.15.0-154-generic   containerd://1.4.9
10.249.0.8   Ready    <none>   19d   v1.20.10+IKS   10.249.0.8    10.249.0.8    Ubuntu 18.04.5 LTS   4.15.0-154-generic   containerd://1.4.9
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# ibmcloud ks cluster addon ls --cluster tor-vpc-test
OK
Name                   Version                Health State   Health Status   
vpc-block-csi-driver   3.0.1* (4.0 default)   normal         Addon Ready. For more info: http://ibm.biz/addon-state (H1500)   
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# ibmcloud ks cluster addon vpc-block-csi-driver  --cluster tor-vpc-test
FAILED
'vpc-block-csi-driver' is not a registered command. See 'ibmcloud ks cluster addon help'.

root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# ibmcloud ks cluster addon disable vpc-block-csi-driver  --cluster tor-vpc-test
Data and resources that you created for the add-on might be deleted when the add-on is disabled. Continue? [y/N]> y
Disabling add-on vpc-block-csi-driver for cluster tor-vpc-test...
OK
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# kubectl get pods -n kube-system | grep block
ibm-vpc-block-csi-controller-0                      0/6     Terminating   0          19d
ibm-vpc-block-csi-node-w9wv4                        0/4     Terminating   0          19d
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# kubectl get pods -n kube-system | grep block
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# 
```


```
# kubectl get pods -n kube-system | grep block
ibm-vpc-block-csi-controller-0                      6/6     Running   0          25s
ibm-vpc-block-csi-node-jb8gl                        4/4     Running   0          24s
ibm-vpc-block-csi-node-kj2fg                        4/4     Running   0          24s
ibm-vpc-block-csi-node-lbz65                        4/4     Running   0          24s
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# kubectl describe pod -n kube-system ibm-vpc-block-csi-controller-0  
Name:         ibm-vpc-block-csi-controller-0
Namespace:    kube-system
Priority:     0
Node:         10.249.0.6/10.249.0.6
Start Time:   Mon, 20 Sep 2021 02:51:01 -0700
Labels:       app=ibm-vpc-block-csi-driver
              controller-revision-hash=ibm-vpc-block-csi-controller-dc74fddd6
              statefulset.kubernetes.io/pod-name=ibm-vpc-block-csi-controller-0
Annotations:  cni.projectcalico.org/podIP: 
              cni.projectcalico.org/podIPs: 
              kubernetes.io/psp: ibm-privileged-psp
              prometheus.io/path: /metrics
              prometheus.io/port: 9080
              prometheus.io/scrape: true
Status:       Running
IP:           172.17.43.17
IPs:
  IP:           172.17.43.17
Controlled By:  StatefulSet/ibm-vpc-block-csi-controller
Containers:
  csi-provisioner:
    Container ID:  containerd://9c3d94e55d0f53deaad8f90fd29fe6f2876e8dee1abc99683b61ced14cf50329
    Image:         quay.io/k8scsi/csi-provisioner:v1.6.0
    Image ID:      quay.io/k8scsi/csi-provisioner@sha256:78e3393f5fd5ff6c1e5dada2478cfa456fb7164929e573cf9a87bf6532730679
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=$(ADDRESS)
      --timeout=600s
      --feature-gates=Topology=true
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:03 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
  csi-attacher:
    Container ID:  containerd://f991fb97526d178dac7344cabf37e3252d42f73705f0caeb4f2db482fb417c70
    Image:         quay.io/k8scsi/csi-attacher:v2.2.0
    Image ID:      quay.io/k8scsi/csi-attacher@sha256:2ffa647e8107cfd39e5f464e738dce014c9f5e51b108da36c3ab621048d0bbab
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=/csi/csi.sock
      --timeout=900s
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:03 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:        10m
      memory:     20Mi
    Environment:  <none>
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
  liveness-probe:
    Container ID:  containerd://31d2e4acfe6287cbd7dee76ed52c6b422ea92e0849c613969821edb08b2d7736
    Image:         quay.io/k8scsi/livenessprobe:v2.0.0
    Image ID:      sha256:5d36e6c4d1c2a3dd43452149e63c4a8b619b3c93ee546bcd1f6c9da9efceeeb6
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=/csi/csi.sock
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:03 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     50m
      memory:  50Mi
    Requests:
      cpu:        5m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
  iks-vpc-block-driver:
    Container ID:  containerd://de64b87d803a90bea1872a347fbced4e04ee833d28df0719a2f1ef9dce70a0e7
    Image:         ambikanair/ibm-vpc-block-csi-driver:nilfix
    Image ID:      docker.io/ambikanair/ibm-vpc-block-csi-driver@sha256:618799307d2b0dd3a0ea52ef8a83dacc15e681d2d6becc4798006bf58e01471e
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --v=5
      --endpoint=$(CSI_ENDPOINT)
      --lock_enabled=false
      --sidecarEndpoint=$(SIDECAREP)
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:04 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     500m
      memory:  500Mi
    Requests:
      cpu:     50m
      memory:  100Mi
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=10s #success=1 #failure=5
    Environment Variables from:
      ibm-vpc-block-csi-configmap  ConfigMap  Optional: false
    Environment:
      SIDECAREP:      /csi/provider.sock
      POD_NAME:       ibm-vpc-block-csi-controller-0 (v1:metadata.name)
      POD_NAMESPACE:  kube-system (v1:metadata.namespace)
    Mounts:
      /csi from socket-dir (rw)
      /etc/storage_ibmc from customer-auth (ro)
      /etc/storage_ibmc/cluster_info from cluster-info (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
  csi-resizer:
    Container ID:  containerd://0f0507dc3355ecbda5dcfb79b1c16475f6392f5c65b85e800d018c67b08ca476
    Image:         quay.io/k8scsi/csi-resizer:v1.0.0
    Image ID:      quay.io/k8scsi/csi-resizer@sha256:83655a67f84ac500b67fd0e7084d1eaa9828321d179269171aa0c9bc35ca8ef9
    Port:          <none>
    Host Port:     <none>
    Args:
      --v=5
      --csi-address=$(ADDRESS)
      --timeout=600s
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:04 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:  /csi/csi.sock
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
  storage-secret-sidecar:
    Container ID:  containerd://39353c2a78d198992fd5ad7fd088ac133b76d94afa6294bdb5f994636a044211
    Image:         icr.io/obs/armada-storage-secret:v1.1.1
    Image ID:      icr.io/obs/armada-storage-secret@sha256:da3078d0e445e8bacc1a10f0247998a86d091afe8a4b538b41b54ee5daedccc2
    Port:          <none>
    Host Port:     <none>
    Args:
      --endpoint=$(ENDPOINT)
    State:          Running
      Started:      Mon, 20 Sep 2021 02:51:05 -0700
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     10m
      memory:  20Mi
    Requests:
      cpu:     3m
      memory:  5Mi
    Environment:
      ENDPOINT:            unix:/sidecardir/provider.sock
      SECRET_CONFIG_PATH:  /etc/storage_ibmc/cluster_info
    Mounts:
      /etc/storage_ibmc/cluster_info from cluster-info (ro)
      /sidecardir from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ibm-vpc-block-controller-sa-token-rv2d8 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  socket-dir:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  customer-auth:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  storage-secret-store
    Optional:    false
  cluster-info:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cluster-info
    Optional:  false
  ibm-vpc-block-controller-sa-token-rv2d8:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ibm-vpc-block-controller-sa-token-rv2d8
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 600s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 600s
Events:
  Type    Reason     Age    From               Message
  ----    ------     ----   ----               -------
  Normal  Scheduled  3m39s  default-scheduler  Successfully assigned kube-system/ibm-vpc-block-csi-controller-0 to 10.249.0.6
  Normal  Pulling    3m38s  kubelet            Pulling image "quay.io/k8scsi/csi-provisioner:v1.6.0"
  Normal  Pulled     3m38s  kubelet            Successfully pulled image "quay.io/k8scsi/csi-provisioner:v1.6.0" in 165.868628ms
  Normal  Created    3m38s  kubelet            Created container csi-provisioner
  Normal  Created    3m37s  kubelet            Created container iks-vpc-block-driver
  Normal  Pulled     3m37s  kubelet            Container image "quay.io/k8scsi/csi-attacher:v2.2.0" already present on machine
  Normal  Created    3m37s  kubelet            Created container csi-attacher
  Normal  Started    3m37s  kubelet            Started container csi-attacher
  Normal  Pulled     3m37s  kubelet            Container image "quay.io/k8scsi/livenessprobe:v2.0.0" already present on machine
  Normal  Created    3m37s  kubelet            Created container liveness-probe
  Normal  Started    3m37s  kubelet            Started container liveness-probe
  Normal  Started    3m37s  kubelet            Started container csi-provisioner
  Normal  Pulled     3m37s  kubelet            Successfully pulled image "ambikanair/ibm-vpc-block-csi-driver:nilfix" in 324.382171ms
  Normal  Pulling    3m37s  kubelet            Pulling image "ambikanair/ibm-vpc-block-csi-driver:nilfix"
  Normal  Started    3m36s  kubelet            Started container iks-vpc-block-driver
  Normal  Pulling    3m36s  kubelet            Pulling image "quay.io/k8scsi/csi-resizer:v1.0.0"
  Normal  Pulled     3m36s  kubelet            Successfully pulled image "quay.io/k8scsi/csi-resizer:v1.0.0" in 173.831558ms
  Normal  Created    3m36s  kubelet            Created container csi-resizer
  Normal  Started    3m36s  kubelet            Started container csi-resizer
  Normal  Pulling    3m36s  kubelet            Pulling image "icr.io/obs/armada-storage-secret:v1.1.1"
  Normal  Pulled     3m36s  kubelet            Successfully pulled image "icr.io/obs/armada-storage-secret:v1.1.1" in 415.026525ms
  Normal  Created    3m35s  kubelet            Created container storage-secret-sidecar
  Normal  Started    3m35s  kubelet            Started container storage-secret-sidecar
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# 
```

ambiknai commented:
```
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# bash deploy/kubernetes/driver/kubernetes/deploy-vpc-block-driver.sh stage
This will install 'stage' version of vpc csi driver!
serviceaccount/ibm-vpc-block-controller-sa unchanged
serviceaccount/ibm-vpc-block-node-sa unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-driver-registrar-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-external-attacher-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-external-resizer-role unchanged
clusterrole.rbac.authorization.k8s.io/vpc-block-provisioner-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-driver-registrar-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-external-attacher-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-external-resizer-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/vpc-block-provisioner-binding unchanged
configmap/ibm-vpc-block-csi-configmap unchanged
statefulset.apps/ibm-vpc-block-csi-controller created
daemonset.apps/ibm-vpc-block-csi-node created
Warning: storage.k8s.io/v1beta1 CSIDriver is deprecated in v1.19+, unavailable in v1.22+; use storage.k8s.io/v1 CSIDriver
csidriver.storage.k8s.io/vpc.block.csi.ibm.io unchanged
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# vi examples/kubernetes/
README.md                   my-storagesecretclass.yaml  raw-block-pod.yaml          storageclass-secret.yaml    validPVC.yaml               
invalidPVC.yaml             pvc-secret.yaml             raw-block-pvc.yaml          validPOD.yaml               
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# vi examples/kubernetes/
README.md                   my-storagesecretclass.yaml  raw-block-pod.yaml          storageclass-secret.yaml    validPVC.yaml               
invalidPVC.yaml             pvc-secret.yaml             raw-block-pvc.yaml          validPOD.yaml               
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# vi examples/kubernetes/validP
validPOD.yaml  validPVC.yaml  
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# vi examples/kubernetes/validP
validPOD.yaml  validPVC.yaml  
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# vi examples/kubernetes/validPVC.yaml 
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc create -f examples/kubernetes/validPVC.yaml 
persistentvolumeclaim/csi-block-pvc-three created
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc get pvc
NAME                  STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                AGE
csi-block-pvc-good    Bound     pvc-d3a2f013-7f25-4c4b-bc03-5e7bbc920854   10Gi       RWO            ibmc-vpc-block-5iops-tier   94m
csi-block-pvc-one     Bound     pvc-ff04f975-d5d4-4876-8f5f-57543ba70ea6   10Gi       RWO            ibmc-vpc-block-5iops-tier   75m
csi-block-pvc-three   Pending                                                                        ibmc-vpc-block-5iops-tier   7s
csi-block-pvc-two     Bound     pvc-ef5f7bd4-d668-49e4-b09d-46a93c0c1971   10Gi       RWO            ibmc-vpc-block-5iops-tier   45m
www-web-0             Bound     pvc-bbadf8f7-eaa9-466a-a428-7f609fb5258b   25Gi       RWO            ibmc-vpc-block-5iops-tier   5d2h
www-web-1             Bound     pvc-4ab28b5d-758a-45de-81d5-4050cda0270e   25Gi       RWO            ibmc-vpc-block-5iops-tier   5d2h
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc describe pvc csi-block-pvc-three
Name:          csi-block-pvc-three
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Mounted By:    <none>
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc describe pvc csi-block-pvc-three
Name:          csi-block-pvc-three
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Bound
Volume:        pvc-0d09da97-9007-4884-a7dd-e33d4a0fd54d
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    <none>
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc describe pvc csi-block-pvc-three
Name:          csi-block-pvc-three
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Bound
Volume:        pvc-0d09da97-9007-4884-a7dd-e33d4a0fd54d
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    <none>
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc get pvc
NAME                  STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                AGE
csi-block-pvc-good    Bound    pvc-d3a2f013-7f25-4c4b-bc03-5e7bbc920854   10Gi       RWO            ibmc-vpc-block-5iops-tier   95m
csi-block-pvc-one     Bound    pvc-ff04f975-d5d4-4876-8f5f-57543ba70ea6   10Gi       RWO            ibmc-vpc-block-5iops-tier   76m
csi-block-pvc-three   Bound    pvc-0d09da97-9007-4884-a7dd-e33d4a0fd54d   10Gi       RWO            ibmc-vpc-block-5iops-tier   51s
csi-block-pvc-two     Bound    pvc-ef5f7bd4-d668-49e4-b09d-46a93c0c1971   10Gi       RWO            ibmc-vpc-block-5iops-tier   46m
www-web-0             Bound    pvc-bbadf8f7-eaa9-466a-a428-7f609fb5258b   25Gi       RWO            ibmc-vpc-block-5iops-tier   5d2h
www-web-1             Bound    pvc-4ab28b5d-758a-45de-81d5-4050cda0270e   25Gi       RWO            ibmc-vpc-block-5iops-tier   5d2h
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# oc describe pvc csi-block-pvc-three
Name:          csi-block-pvc-three
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Bound
Volume:        pvc-0d09da97-9007-4884-a7dd-e33d4a0fd54d
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    <none>
root@devmachine1:~/WORKSPACE/src/github.ibm.com/alchemy-containers/ibm-vpc-block-csi-driver# 
```

ambiknai commented:
GHE: https://github.ibm.com/alchemy-containers/armada-storage/issues/2428
